packages/paste-core/components/switch/src/SwitchKnob.tsx

```diff
@@ -1,30 +1,39 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import type {SwitchKnobProps} from './types';
 import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
 import {CheckboxCheckIcon} from '@twilio-paste/icons/esm/CheckboxCheckIcon';
 import {useTheme} from '@twilio-paste/theme';
 
-const getSwitchKnobTransformValue = (disabled: boolean, switchIsOn: boolean, isHovering: boolean): string => {
+const getSwitchKnobTransformValue = (disabled: boolean, checked: boolean, isHovering: boolean): string => {
   if (!disabled) {
-    if (!switchIsOn && isHovering) return 'translateX(5%)';
-    if (switchIsOn && !isHovering) return 'translateX(100%)';
-    if (switchIsOn && isHovering) return 'translateX(95%)';
+    if (!checked && isHovering) return 'translateX(5%)';
+    if (checked && !isHovering) return 'translateX(100%)';
+    if (checked && isHovering) return 'translateX(95%)';
   } else if (disabled) {
-    if (!switchIsOn) return 'translate(0%)';
-    if (switchIsOn) return 'translateX(100%)';
+    if (!checked) return 'translate(0%)';
+    if (checked) return 'translateX(100%)';
   }
   return 'translate(0%)';
 };
 
+export interface SwitchKnobProps {
+  element: BoxProps['element'];
+  disabled?: boolean;
+  checked?: boolean;
+  isHovering: boolean;
+  height: string;
+  hasError?: boolean;
+}
+
 const SwitchKnob = React.forwardRef<HTMLDivElement, SwitchKnobProps>(
-  ({element, disabled = false, switchIsOn, isHovering, height, ...props}, ref) => {
+  ({element = 'SWITCH', disabled = false, checked = false, isHovering, height, hasError = false, ...props}, ref) => {
```

Comment:
It's called SWITCH here because there are two child elements, so it concats below in the same file to `SWITCH_KNOB` and `SWITCH_ICON`. I kind of prefer this pattern because you pass in one element everywhere and all the children handle adding the suffix.